<div style="border:solid Chocolate 2px; padding: 40px">

<b>Привет!👋</b>

Меня зовут Огурлиев Анзор, я буду ревьюером твоего проекта. Сразу хочу предложить тебе общаться на "ты", если тебе это неудобно, скажи мне об этом. Нам предстоит пройти с тобой через тернистый и от того интересный путь анализа данных! Можешь не переживать, я всегда помогу тебе, если у тебя возникнут вопросы)

По ходу твоей работы я буду оставлять свои комментарии, по выполненным тобой шагам. Попрошу тебя не удалять их, так мне будет проще отслеживать твой прогресс в случае повторной проверки. 
    
Очень важно вести диалог и обсуждать те пункты, которые у тебя могли не получиться, поэтому, ты можешь оставлять свои комментарии, и мы обязательно во всем разберемся! Самое главное - выделяй свои комментарии удобным для тебя способом (цвет, рамка), но так, чтобы они отчетлитво отличались от моих комментариев. Так мы ни в чем не запутаемся)

Свои комментарии я буду выделять так: <font color='green'>зеленый</font>, <font color='gold'>желтый</font> и <font color='red'>красный</font> цвет, например:

<br/>

<div class="alert alert-success">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Этот комментарий я буду оставлять, если твое решение оптимально для данного шага.
</div>

<br/>

<div class="alert alert-warning">
    <h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> Это не означает, что ты совершил ошибку. Такими комментариями я буду отмечать места, где ты справился с задачей, но её можно решить ещё лучше. Также в таких комментариях я буду оставлять для тебя другие советы.
</div>


<br/>
<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера V1<a class="tocSkip"></h2>

    
<b>На доработку🤔:</b>
Если решение, которое ты применил на определенном шаге, требует значительных изменений, то я буду оставлять такой комментарий. Тебе нужно будет подумать еще раз как правильным образом решить данный этап. Если ревью содержит такие комментарии, то я не смогу принять проект с первого раза.
</div>
    
P.S. В оглавлении комментария ты увидишь `V1` (`V1`, `V2`, `V3`) - это версия моего комментария, которая понадобится нам с тобой, чтобы не запутаться, если потребуется оставить тебе комментарии после первой проверки.

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [31]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   

       family_status       family_status_id gender income_type  debt  \
0    женат / замужем        женат / замужем      F   сотрудник     0   
1    женат / замужем        женат / замужем      F   сотрудник     0   
2    женат / замужем        женат / замужем      M   сотрудник     0   
3    женат / замужем        женат / замужем      M   сотрудник     0   
4   гражданский брак       гражданский брак      F   пенсионер     0   
5   гражданский брак       гражданский брак      M   компаньон     0   
6    женат / замужем        женат / замужем      F   компаньон     0   
7    женат / замужем        женат / замужем      M   сотрудник     0   
8   гражданский брак       гражданский брак      F   сотрудник     0   
9    женат / замужем        женат / замужем      M   сотрудник     0   
10   женат / замужем        женат / замужем      M   компаньон     0   
11   женат / замужем        женат / замужем      F   сотрудник     0   
12  гражданский брак       гражданский брак      M   пенсионер     0   
13   женат / замужем        женат / замужем      F   сотрудник     0   
14  гражданский брак       гражданский брак      F   компаньон     1   
15   женат / замужем        женат / замужем      F   сотрудник     0   
16   женат / замужем        женат / замужем      F   сотрудник     0   
17  гражданский брак       гражданский брак      M   сотрудник     0   
18    вдовец / вдова  Не женат / не замужем      F   пенсионер     0   
19         в разводе              в разводе      F   сотрудник     0   

    total_income                                 purpose  \
0         253875                           покупка жилья   
1         112080                 приобретение автомобиля   
2         145885                           покупка жилья   
3         267628              дополнительное образование   
4         158616                         сыграть свадьбу   
5         255763                           покупка жилья   
6         240525                       операции с жильем   
7         135823                             образование   
8          95856                   на проведение свадьбы   
9         144425                 покупка жилья для семьи   
10        113943                    покупка недвижимости   
11         77069       покупка коммерческой недвижимости   
12        118514                         сыграть свадьбу   
13        130458                 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# рассчитаем долю невозвратных кредитов в зависимости от количества детей:
print(data.pivot_table(index=['children'], values='debt', aggfunc=['sum','count','mean']))

           sum  count      mean
          debt   debt      debt
children                       
0         1063  14091  0.075438
1          444   4808  0.092346
2          194   2052  0.094542
3           27    330  0.081818
4            4     41  0.097561
5            0      9  0.000000


<div class="alert alert-success">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Правильно построил сводную таблицу, молодец!
</div>

**Вывод:** Анализируя данные, можно сделать вывод, что максимальный процент невозвратных кредитов наблюдается у группы с четырьмя (4 ребенка) детьми, а минимальный у группы с пятью (5 детей) детьми  при этом нельзя говорить, что это полностью репрезентативно т.к. для утверждения наличия прямой зависимости слишком представлена незначительная выборка в данных группах. В группах с одним или двумя детьми доля пневозвратных кредитов практически идентичная - 9,23 % и 9,45 %, при этом репрезентативность и валидность результатов здесь намного выше, т.к. выборка в группе с одним ребенков больше, чем в группе с четырься детьми в 120 раз(4808 случаев), а в группе с двумя детьми в 50 раз (2052 случая). У клиентов с 3 детьми  доля невозвратных кредитов меньше, чем в группах с одним и двумя детьми (8, 18 %), у группы лиц с пятью детьми невозвратные кредит отсутствуют. При этом доля невозвратных кредитов у клиентов без детей ниже, чем у всех групп с детьми, кроме категории с наличием пяти детей. Таким образом нельзя говорить о наличии прямой или обратной зависимостей между количеством детей и возврата кредита в срок.

<div class="alert alert-warning">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> Ты верно перечислил особенности разных групп. Отмечу, что ты верон увидел несбалансированность клиентов по группам и тот факт, что в некоторых группах клиентов недостаточно для корректного анализа, это касается и группы с 3 детьми. Нужно выявить зависимость только по группам с достаточным количестов клиентов.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [33]:
# рассчитаем процент невозвратных кредитов в зависимости от семейного положения:
# для этого к в столбце 'family_status_id' имеющиеся цифры на понятия в соответствии таблицей соответствия, где :0:'женат / замужем', 1:'гражданский брак',2:'вдовец / вдова', 3:'в разводе',4:'Не женат / не замужем'
data['family_status_id'] = data['family_status_id'].replace({0:'женат / замужем', 1:'гражданский брак', 2:'вдовец / вдова', 3:'в разводе', 4:'Не женат / не замужем'})
# рассчитаем долю невозвратных кредитов в зависимости от семейного положения:
print(data.pivot_table(index=['family_status_id'], values='debt', aggfunc=['sum','count','mean']))

                       sum  count      mean
                      debt   debt      debt
family_status_id                           
Не женат / не замужем   63    951  0.066246
в разводе               84   1189  0.070648
вдовец / вдова         273   2796  0.097639
гражданский брак       385   4134  0.093130
женат / замужем        927  12261  0.075606


<div class="alert alert-success">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Видно, что ты хорошо справляешься с методами построения сводных таблиц, так держать!
</div>

**Вывод:** Анализ данных показал,что  наибольший процент невозвратных кредитов в группах с семейным положением "вдовец / вдова"и "гражданский брак" (9,76 % и 9,31 % соответственно). Это можно объяснить возможными финансовыми проблемами у вдов и вдовцов в связи с потерей партнера, а в группе живущей в гражданском браке(что с терминологической и юридической стороны является настощим браком, но заменяет в обиходе термин сожительство), возможно, отсутствует планирование общего бюджета . У клиентов с семейным положением "не женат" наиболее низкий процент невозвратных кредитов - 6,66 % .

<div class="alert alert-success">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Все верно, можно сделать вывод, что клиенты, побывавшие в официальном браке, реже допускают просрочки по кредитам. Гражданский брак сюда не относится, если я правильно помню, то это просто сожительство.
</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [27]:
# рассчитаем долю не возврата кредита в зависимости от уровня дохода:
print(data.pivot_table(index=['total_income_category'], values='debt', aggfunc=['sum','count','mean']))

                        sum  count      mean
                       debt   debt      debt
total_income_category                       
A                         2     25  0.080000
B                       354   5014  0.070602
C                      1353  15921  0.084982
D                        21    349  0.060172
E                         2     22  0.090909


**Вывод:** В ходе анализа данных удалось выявить,что , максимальный процени невозвратных кредитов у клиентов с уровнем дохода в категории E (0–30000) - 9.09 % (Возможно людям банально не хватает денег на жизнь и присутвтвуют кредиты в других банках, которые идут на погашение других кредитов).Самый низкий процент невозвратных кредитов у клиентов с уровнем дохода категории D (30 001–50 000) - 6.02 %. У клиентов с уровнем дохода категории   C (50 001–200 000) и A (выше 1 000 001) приблизительно одинаковый % не возврата (  8,49 % и 8,00 %соответственно)(Растут возможности кредитоваться, в особенности в категории , которая C входит в " нижний средний класс", который зарабатывет почти как "высший низший", а старается тратить как "низший высший").
UPD: более тщательный взгляд на данные позволяет сделать вывод, что выборка по группам доходов Е (0–30000) и A (выше 1 000 001), не позволяет полноценно их учитывать, для формирования вывовода о наличии зависимости между уровнем дохода и возвратностью кредита, т.к. это 22 и 25 клиентов соответственно. Группа D (30001-50 000) обладает сравнительно большим диапазоном данных, но все же недостаточным для равноценного соспоставления с группами В(200 001-1000 000) и С (50001-200 000), где не смотря на наличие значительного разрыва в доходах  не наблюдается сопоставимого разрыва в возвратности полученных кредитов. Таким образом, исходя из имеющихся данных нельзя сделать достоверный вывод о наличии зависимости между уровнем дохода и возвратностью кредита.
upd2:не смотря на наличие большого разрыва в доходах (до 5кратного разрыва), не наблюдается четкой зависимости с возратностью кредита(хотя в группне B невозратных кредитов на 1,5% меньше, что на больших цифрах(миллионах и миллиардах и на таймфрейме в десятки лет) необходимо также учитывать, но это не было целью нашей работы

<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера V1<a class="tocSkip"></h2>

    
<b>На доработку🤔:</b>
Обрати внимание на распределение клиентов по группам. Можно ли учитывать все группы при выявлении зависимости?
    
Подсказка: зависимость нужно определять только по 2 группам с достаточным количеством клиентов.
</div>

<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера V2<a class="tocSkip"></h2>

    
<b>На доработку🤔:</b>
Что в твоем понимании значительный разрыв? Разница в 1-2% может быть значимой, нельзя отвергать её без дополнительных проверок. Также при пересчете в абсолютных значениях разница между просрочками может составлять миллионы рублей, например.
</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера V3<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Почему это не было целью нашей работы? Нужно выяснить влияние уровня дохода на просрочки, при этом нужно учитывать все особенности датафрейма (несбалансированность выборки, разница в 1.5% и тд.)

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [28]:
# рассчитаем долю не возврата кредита в зависимости от цели кредита:
print(data.pivot_table(index=['purpose_category'], values='debt', aggfunc=['sum','count','mean']))# Ваш код будет здесь. Вы можете создавать новые ячейки.

                          sum  count      mean
                         debt   debt      debt
purpose_category                              
операции с автомобилем    400   4279  0.093480
операции с недвижимостью  780  10751  0.072551
получение образования     369   3988  0.092528
проведение свадьбы        183   2313  0.079118


**Вывод:** При анализе данных выявлено, максимальный процент не возврата кредита у клиентов с целями  кредитования на "получение образования" и "операции с автомобилем" и  ( 9,25 % и  9,34 %соответственно). Это может быть связано с поиском работы и нестабильным заработком у категории "на образование" и с риском повреждения имущества и тратами на его эксплуатацию у категории "на операции с автомобилем". Далее идет группа клиентов, которые берут кредит для "проведения свадьбы" - 7,91 % невозвращенных кредитов. более низкий процент может быть связан с тем, что на свадьбу иногда дарят деньги, которые идут на погашения кредита, но наличие невозвратных кредитов, говорит о том, что деньги дарят не всегда. Самый низкий процент не возврата у клиентов с целью кредита "операции с недвижимостью"- 7,25 %, т.к. в данной категории, проверки должны осуществляться тщательнее, а рисковое обеспечение быть достаточным.

<div class="alert alert-success">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Именно так, все верно.
</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Пропыщенные значения в стаже работы "days_employed" могут быть связаны с утратами трудовых книжек т.к. еще не все оцифровано, а некоторые скрывают причину увольнения, поэтому не всегда указывают данные с места предыдущей работы. Пропущенные значения в колонке "total_income" могут быть связаны с невнимательностью или корыстью сотрудников, желающих сделать план и продать больше кредитов( так и случился мировой финансовый кризис в 2008 г.).

<div class="alert alert-success">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Эти причины могут возникнуть в реальной жизни, согласен с тобой.
</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* медианные значения более репрезентативны, чем средние, т.к. они учитывают не только разброс, но и выборку( количество людей в группе)
UPD:соглашусь за замечанием, что выбросы, которые сильно повлияют на конечный результат при расчете по методу средних чисел,  медианный расчет позволяет нивелировать числовые аномалии(для примера рост ВВП Монголии или Ирландии, где есть значительные аномалии в отдельные годы)

<div class="alert alert-warning">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> Не совсем корректно ты сформулировал вывод. Скорее используют медиану, потому что она меньше зависит от выбросов в выборке.

<div class="alert alert-success">
<h2> Комментарий ревьюера V2<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Именно так, все верно.
</div>

### Шаг 4: общий вывод.

Напишите ваш общий вывод. При проведении анализа не было выявлено прямой зависимости между количеством детей и процентом невозвратных кредитов, но при этом есть незначительная зависимость между семейным статусом и возможностью вернуть кредит(она меньше у вдовствующих и тех, кто не оформляет брак, а живет гражданским браком). Большая зависимость  прослеживается от уровня дохода, что и обуславливает возможность кредитоваться и возвращать кредит. Логическая взаимосвязь прослеживается при обработке по целям кредитования, но здесь необходимо проведения более глубоко исследования с более широким набором данных.
UPD: повторное рассмотрение позволило доработать положения итогового и сказать о том что:
-для проведения детального и репрезентативного исследования необходимо иметь широкую выборку по различным категориям, в противном случае возможно выявление ложных закономерностей или выводов;
-средние значения менее репрезентативны, чем медианные, т.к. позволяют нивелировать влияние аномальных выбросов по отдельным показателям;
-исходя из имеющихся данных нельзя с полной уверенностью сказать что есть прямая зависимость между долей невозвратных кредитов и уровнем дохода, а также между долей невозвратных кредитов и количеством детей;
-при этом есть определенные закономерности между семейным статусом и возратностью кредита,она меньше у вдовствующих и тех, кто не оформляет брак, а живет гражданским браком(сожительствует)
-можно сделать предположение о зависимости цели кредитования и перспектив его успешного погашения(кредиты на операции с автомобилем и образование возвращаются меньше, чем на свадьбу и недвижимость) т.к. для разных целей применяются разные по строгости финансовые проверки (комплаенс) и требуют разного уровня обеспечения кредита, что и обуславливают различную степень риска возвратности заемных средств.
UPD2: при анализи зависимости  количества детей и процента невозвтраных кредитов, нельзя проследить четкой взаимосвязи, в связи с неравномерным распределением клиентом по рассматриваемым группам: клиентов без детей 14091, клиентов с одним ребенком уже 3,5 раза меньше- 4808, а с двумя детьми в 2,4 раза меньше, чем клиентов с одним ребенком и почти в 7 раз меньше, чем клиентов без детей. Разрыв с оставшимися  группами еще больше. Поэтому нельзя сопоставлять данные группы, т.к. можно сделать ошибочные выводы о том, что самые надежные клиенты имеют 5 детей, за ними следую бездетные, а группы наибольшего риска имеют четыре и два ребенка. Но данный вывод не имеет под собой достаточной статистической базы.


<div class="alert alert-warning">
<h2> Комментарий ревьюера V1<a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> В итоговом выводе нужно достаточно подробно написать основные пункты выявленных зависимостей. Стоит доработать вывод с учетом моих комментариев по ходу работы.

<div class="alert alert-warning">
<h2> Комментарий ревьюера V2<a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> Нужно дорабоать вывод о зависимости от количества детей и от уровня дохода.

<div class="alert alert-success">
<h2> Комментарий ревьюера V3<a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> Отличный вывод, все четко, понятно и структурировано. Вывод в полной мере отражает существующие зависимости, и по нему можно принимать дальнейшие решения.
</div>

<div style="border:solid Chocolate 2px; padding: 40px">

**Общий вывод по проекту**

Спасибо за твой проект! Я уверен, что ты приложил все усилия, чтобы сделать этот проект. Это видно по твоей работе, а сформулированные выводы отражают, насколько успешно ты осваиваешь материал. 

**Отмечу отдельные положительные моменты проекта🙂**:
 
* Ты полностью следовал всем этапам выполнения задания и справился почти со всем. Хорошая работа!
* Отдельно выделю некоторые выводы, которые ты смог сформулировать абсолютно точно!
* У тебя опрятный и читаемый код. Возможно это звучит банально, однако при работе с чужим кодом это крайне важно)


**На что стоит обратить внимание🤓**:
    
* Важно доработать итоговый вывод.
* Какая особенность данных не позволяет сходу дать ответ на вопрос: "Есть ли зависимость между уровнем дохода и возвратом кредита в срок?"
* Остальные рекомендации ты найдешь в моих комментариях!


**Желаю удачи и жду твой проект на повторное ревью!**😉

<div style="border:solid Chocolate 2px; padding: 40px">

**Общий вывод по проекту**

**На что стоит обратить внимание🤓**:
    
* Важно доработать итоговый вывод.
* Нужно доработать анализ зависимости в пункте с количеством детей и уровнем дохода.
* Остальные рекомендации ты найдешь в моих комментариях!


**Желаю удачи и жду твой проект на повторное ревью!**😉

<div style="border:solid Chocolate 2px; padding: 40px">

Ты отлично справился с данной работой! Видно, как ты успешно осваиваешь материал, применяешь полученные знания на практике. Мне было очень приятно проверять твою работу, надеюсь, мои комментарии и советы были для тебя полезными.

**Желаю удачи в последующем обучении и исследованиях!**😉